rllib/core/rl_trainer/torch/torch_rl_trainer.py

```diff
@@ -1,10 +1,185 @@
 import logging
-from ray.rllib.core.rl_trainer.rl_trainer import RLTrainer
+from typing import (
+    Any,
+    Mapping,
+    Union,
+    Type,
+    Sequence,
+    Hashable,
+    Optional,
+    Callable,
+)
+import torch
 
-logger = logging.getLogger(__name__)
+from ray.train.torch.train_loop_utils import _TorchAccelerator
+
+from ray.rllib.core.rl_module.rl_module import RLModule, ModuleID
+from ray.rllib.core.rl_trainer.rl_trainer import (
+    RLTrainer,
+    MultiAgentRLModule,
+    ParamOptimizerPairs,
+    Optimizer,
+    ParamType,
+    ParamDictType,
+)
+from ray.rllib.core.rl_module.torch.torch_rl_module import DDPRLModuleWrapper
+from ray.rllib.policy.sample_batch import MultiAgentBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.typing import TensorType
+from ray.rllib.utils.nested_dict import NestedDict
 
-# TODO: Implement this
+logger = logging.getLogger(__name__)
 
 
 class TorchRLTrainer(RLTrainer):
-    pass
+    def __init__(
+        self,
+        module_class: Union[Type[RLModule], Type[MultiAgentRLModule]],
+        module_kwargs: Mapping[str, Any],
+        scaling_config: Mapping[str, Any],
+        optimizer_config: Mapping[str, Any],
+        distributed: bool = False,
+        in_test: bool = False,
+    ):
+        super().__init__(
+            module_class=module_class,
+            module_kwargs=module_kwargs,
+            scaling_config=scaling_config,
+            optimizer_config=optimizer_config,
+            distributed=distributed,
+            in_test=in_test,
+        )
+
+        self._world_size = scaling_config.get("num_workers", 1)
+        self._use_gpu = scaling_config.get("use_gpu", False)
+        # These attributes are set in the `build` method.
+        self._device = None
+
+    @property
+    @override(RLTrainer)
+    def module(self) -> MultiAgentRLModule:
+        return self._module
+
+    @override(RLTrainer)
+    def configure_optimizers(self) -> ParamOptimizerPairs:
+        lr = self.optimizer_config.get("lr", 1e-3)
```

Comment:
should not have a default lr here. default should be set in AlgorithmConfig.

Reply:
Agreed. Please add a utility "sub-config-compilation"-method to AlgorithmConfig: `get_optimizer_config()`, which can be used to create a new OptimizerConfig (dataclass?) object by `Algorithm` to configure its RLTrainer(s). Please no more dicts as configs, anywhere!

Reply:
ok ok. For now optimizer config is a dict. Please go with it, for first iteration until PoC it's easier to leave it as a dict.

Reply:
but I agree on the no default thing in the proceeding classes.